src/util/alloc/free_list_allocator.rs

```diff
@@ -428,43 +425,45 @@ impl<VM: VMBinding> FreeListAllocator<VM> {
     /// for mark sweep.
     const ABANDON_BLOCKS_IN_RESET: bool = true;
 
+    /// Lazy sweeping. We just move all the blocks to the unswept block list.
     #[cfg(not(feature = "eager_sweeping"))]
     fn reset(&mut self) {
         trace!("reset");
         // consumed and available are now unswept
         for bin in 0..MI_BIN_FULL {
             let unswept = self.unswept_blocks.get_mut(bin).unwrap();
-            unswept.lock();
+
+            // We should have no unswept blocks here. Blocks should be either in avialable, or consumed.
+            debug_assert!(unswept.is_empty());
```

Comment:
This should only be true if `ABANDON_BLOCKS_IN_RESET == true`.  Otherwise, after this GC, one mutator may still have unswept blocks when another mutator has exhausted all available blocks and triggers a GC.

Reply:
Another minor spelling mistake: "avialable" -> "available".